PubSub/src/Subscription.php

```diff
@@ -838,6 +943,127 @@ public function modifyAckDeadlineBatch(array $messages, $seconds, array $options
         }
     }
 
+    /**
+     * Helper that sends a request to modify the ack deadline but with retries.
+     *
+     * @param Message[] $messages An array of messages
+     * @param int $seconds The new ack deadline with respect to the time
+     *        this request was sent to the Pub/Sub system. Must be >= 0. For
+     *        example, if the value is 10, the new ack deadline will expire 10
+     *        seconds after the ModifyAckDeadline call was made. Specifying
+     *        zero may immediately make the message available for another pull
+     *        request.
+     * @param array $options Configuration Options
+     *
+     * @return array
+     */
+    private function modifyAckDeadlineBatchWithRetries(array $messages, $seconds, array $options)
+    {
+        $actionFunc = function (&$messages, $options) use ($seconds) {
+            $this->connection->modifyAckDeadline($options + [
+                'subscription' => $this->name,
+                'ackIds' => $this->getMessageAckIds($messages),
+                'ackDeadlineSeconds' => $seconds
+            ]);
+        };
+
+        return $this->retryEodAction($actionFunc, $messages, $options);
```

Comment:
I am a little concerned about this part. If feels like if the client set `returnFailures` to true, without exactly once delivery, they will take a processing hit, since we will parse the errorInfo and check if it is exactly once. Since PHP uses sync pull, I do not think there is a good way to avoid it. But should we document that this option leads to extra computations?

Reply:
[L 1010](https://github.com/googleapis/google-cloud-php/pull/5315/files#diff-c21815ead183d93472342f6e53240ce6272170ab7cb5e06545af509f939ed44cR1010) has a little mention of this but doesn't talk about computational overheads.

Do you think the computational overhead is enough for us to be documenting it? If the user is using GRPC(GAPIC layer) then the only overhead is unpacking(done by gax, not even in the code of this repo).

If the user is using REST, then the overhead is just looping through the error details to compare the type to get the error info. And, it's not that we are giving the users a choice to circumvent this. So, I wonder if mentioning the overhead of a process that _has to happen_ will do any good here. WDYT?

Reply:
That's fair. Thanks for explaining.